### DISCUSSION TITLE
Handling configurations that depend on PHP version

### DISCUSSION THREAD

**Discussion by julienfalque** [PHP-CS-Fixer/PHP-CS-Fixer#7919] [RFC]
Currently, when a fixer option only makes sense on a subset of supported PHP versions, we throw an exception in case the option is enabled on an incompatible version.

This requires the consumer (end user or ruleset) to make sure the provided configuration is compatible. However, in most cases, the fixer could accept the configuration and simply discard incompatible changes it would lead to.

As an example, with option `elements` of rule `trailing_comma_in_multiline` ([ref](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/7825)), we could always allow `parameters` and skip those on PHP 7.4. Same for `match`, which would likely not even require code changes as the token simply does not exist on PHP 7.4.

**Comment by keradus** [comment]
My suggestion would be (but probably requires review and adjustments of rules)
- warn (or even block by default) user if they are running tool using too new engine (ref https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7312 ) - as pre-requirement
- discard incompatible changes (eg user enabled `parameters`, but fixer disables it due to too low PHP runtime)

**Reply by keradus** [reply]
Closed by https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/pull/8167

**Comment by ghost** [comment]
I shared my opinion in the releated discussion https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/discussions/7312#discussioncomment-9099638
